src/draw/lv_draw_image.c

```diff
@@ -116,16 +114,52 @@ void lv_draw_image(lv_layer_t * layer, const lv_draw_image_dsc_t * dsc, const lv
         return;
     }
 
-    lv_draw_task_t * t = lv_draw_add_task(layer, coords);
-    t->draw_dsc = new_image_dsc;
-    t->type = LV_DRAW_TASK_TYPE_IMAGE;
+    if(new_image_dsc->header.flags & LV_IMAGE_FLAGS_CUSTOM_DRAW) {
+        lv_image_decoder_dsc_t decoder_dsc;
+        res = lv_image_decoder_open(&decoder_dsc, new_image_dsc->src, NULL);
+        if(res != LV_RESULT_OK) {
+            LV_LOG_ERROR("Failed to open image");
+            return;
+        }
 
-    lv_image_buf_get_transformed_area(&t->_real_area, lv_area_get_width(coords), lv_area_get_height(coords),
-                                      dsc->rotation, dsc->scale_x, dsc->scale_y, &dsc->pivot);
-    lv_area_move(&t->_real_area, coords->x1, coords->y1);
+        if(decoder_dsc.decoder && decoder_dsc.decoder->custom_draw_cb) {
+            lv_area_t draw_area = layer->buf_area;
+            int32_t xpos = coords->x1 - draw_area.x1;
+            int32_t ypos = coords->y1 - draw_area.y1;
+
+            lv_area_t coords_area = *coords;
+            lv_area_move(&coords_area, -(coords->x1 - xpos), -(coords->y1 - ypos));
+
+            layer->_clip_area = coords_area;
+#if LV_USE_MATRIX
+            lv_matrix_t matrix;
+            lv_matrix_identity(&matrix);
+            lv_matrix_translate(&matrix, dsc->pivot.x, dsc->pivot.y);
+            lv_matrix_rotate(&matrix, dsc->rotation / 10.0f);
+            lv_matrix_scale(&matrix, dsc->scale_x / 255.0f, dsc->scale_y / 255.0f);
+            lv_matrix_translate(&matrix, -dsc->pivot.x, -dsc->pivot.y);
+
```

Comment:
It can be placed in `custom_draw_cb` for processing at the lower level, and provides a helper function for converting `img_dsc` to matrix. You can refer to the `lv_vg_lite_image_matrix` function.

It can reduce the `#if MATRIX` macro judgment branches, making it easier to maintain and test later.

Reply:
done